app/src/main/java/com/parking_system_kotlin/activities/ReservationActivity.kt

```diff
@@ -0,0 +1,47 @@
+package com.parking_system_kotlin.activities
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import com.parking_system_kotlin.listeners.ListenerDateTime
+import com.parking_system_kotlin.mvp.contracts.ParkingReservationContract
+import com.parking_system_kotlin.mvp.presenter.ParkingReservationPresenter
+import com.parking_system_kotlin.mvp.view.ParkingReservationView
+import com.parkingsystemkotlin.databinding.ActivityReservationBinding
+import java.util.*
+
+class ReservationActivity : AppCompatActivity(), ListenerDateTime {
+
+    private lateinit var binding: ActivityReservationBinding
+    private lateinit var presenter: ParkingReservationContract.ParkingReservationPresenter
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivityReservationBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+        presenter = ParkingReservationPresenter(ParkingReservationView(this, binding))
+
+        setListeners()
+    }
+
+    fun setListeners() {
+        binding.editTextReservationActivityEntry.setOnClickListener { presenter.showDatePicker(this, true) }
+        binding.editTextReservationActivityExit.setOnClickListener { presenter.showDatePicker(this, false) }
+        binding.buttonReservationActivitySave.setOnClickListener { presenter.saveReservation() }
+    }
+
+    override fun setEntryDate(entryDate: Calendar) {
+        presenter.setEntryDate(entryDate)
+    }
+
+    override fun setExitDate(exitDate: Calendar) {
+        presenter.setExitDate(exitDate)
+    }
+
+    companion object {
+        fun newInstance(context: Context): Intent {
+            return Intent(context, ReservationActivity::class.java)
```

Comment:
instead of using return, you could use = to return the instance in a single line

Reply:
done!